lib/improver/blending/weights.py

```diff
@@ -238,12 +242,575 @@ def process_coord(cube, coordinate, coord_exp_vals=None,
+                coordinate is set equal to None.
+        """
+        new_coord_list = []
+        for coord in coord_list:
+            if (coord.name() == exception_coord_name or
+                    base_cube.coord_dims(coord) ==
+                    base_cube.coord_dims(exception_coord_name)):
+                exception_coord = cube_with_exception_coord.coord(coord.name())
+                if cube_with_exception_coord.coord_dims(coord):
+                    index, = cube_with_exception_coord.coord_dims(coord)
+                    new_coord_list.append((exception_coord, index))
+                else:
+                    new_coord_list.append((exception_coord, None))
+            else:
+                if base_cube.coord_dims(coord):
+                    index, = base_cube.coord_dims(coord)
+                    new_coord_list.append((coord, index))
+                else:
+                    new_coord_list.append((coord, None))
+        return new_coord_list
+
+    def _create_new_weights_cube(self, cube, weights, weights_cube):
+        """Create a cube to contain the output of the interpolation.
+        It is currently assumed that the output weights matches the size
+        of the input cube. This will be true if the only difference between
+        the cube and the weights cube is along the dimension of the
+        self.weighting_coord_name coordinate.
+
+        Args:
+            cube (iris.cube.Cube):
+                Cube containing the coordinate information that will be used
+                for setting up the new_weights_cube.
+            weights (np.ndarray):
+                Weights calculated following interpolation.
+            weights_cube (iris.cube.Cube):
+                Cube containg the weights that will be interpolated to find
+                new weights at the specified points.
+
+        Returns:
+            new_weights_cube (iris.cube.Cube):
+                Cube containing the output from the interpolation. The
+                metadata is updated using the weights_cube metadata.
+        """
+        dim_coords_and_dims = (
+            ChooseWeightsLinearFromCube._create_coord_and_dims_list(
+                weights_cube, cube, weights_cube.dim_coords,
+                self.weighting_coord_name))
+        aux_coords_and_dims = (
+            ChooseWeightsLinearFromCube._create_coord_and_dims_list(
+                weights_cube, cube, weights_cube.aux_coords,
+                self.weighting_coord_name))
+
+        new_weights_cube = iris.cube.Cube(
+            weights, standard_name=weights_cube.standard_name,
+            long_name=weights_cube.long_name, units=weights_cube.units,
+            attributes=weights_cube.attributes,
+            cell_methods=weights_cube.cell_methods,
+            dim_coords_and_dims=dim_coords_and_dims,
+            aux_coords_and_dims=aux_coords_and_dims)
+        return new_weights_cube
+
+    def _interpolate_to_create_weights(self, cube, weights_cube):
```

Comment:
There is an identically named function in the `...FromDict` plugin that does almost identical processing.  This is really confusing!  As far as I can see the only difference is that this one has an option to use a different axis rather than zero.  I think it would actually be clearer (and reduce duplication) to get rid of these wrapper functions and make the calls directly from both `_process` functions.  But as a minimum the docstrings need to clarify what the differences are.